### ISSUE TITLE
React keys warning when component body within let

### ISSUE THREAD

**Issue by lynaghk** [tonsky/rum#33]
Using

``` clojure
[rum "0.2.7" :exclusions [cljsjs/react]]
[cljsjs/react-with-addons "0.12.2-7"]
```

Rum causes a React keys warning when a component render body is within a let.
The minimal test case below:

``` clojure
(defn raw-sablono
  []
  (sablono.core/html
   [:div
    [:span "A"]
    [:span "B"]]))


(defn let-sablono
  []
  (sablono.core/html
   (let [x 1]
     [:div
      [:span "A"]
      [:span "B"]])))


(rum/defc raw-rum
  []
  [:div
   [:span "A"]
   [:span "B"]])


(rum/defc let-rum
  []
  (let [x 1]
    [:div
     [:span "A"]
     [:span "B"]]))


(rum/defc app
  []
  [:div
   (raw-sablono)
   (let-sablono)
   (raw-rum)
   (let-rum)])


(.render js/React (app) (.getElementById js/document "container"))

```

Causes the following warnings in the JavaScript console:

```
Each child in an array should have a unique "key" prop. Check the render method of app. See http://fb.me/react-warning-keys for more information.
Each child in an array should have a unique "key" prop. Check the render method of let-rum. See http://fb.me/react-warning-keys for more information.
```

I'm confused by both warnings since none of this code is using sequences --- everything should compile into direct React calls without intermediate arrays.

I've tried debugging by expanding rum's output in a Clojure REPL:

``` clojure
(macroexpand
   '(rum/defc app
      []
      [:div
       [:span]
       [:span]]))
```

but it blows up:

```
1. Unhandled java.lang.IllegalArgumentException
   Syntax error at ([] [:div [:span] [:span]])

                       rum.clj:   25  rum/parse-defc
                       rum.clj:   31  rum/-defc
                       rum.clj:   54  rum/defc
```

**Comment by atroche** [comment]
I'm having a similar problem.

**Comment by andromorv** [comment]
So, i've spent a couple of hours fighting this issue and lost. But i've found how render method will look like.

For example, for this component

``` clj
(defc my-component < static [props]
  (let [x (:x props)]
    [:div {:class-name "test"}
      [:span "Test"]
      [:span (str "Test" x)]]))
```

render method will look like

``` clj
;; Some brainfuck action here

(fn []
  (this-as this
           (let [state (state this)
                 [dom next-state] ((fn [state]
                                     [(apply (fn [props]
                                               (do (let [x (:x props)]
                                                     (js/React.createElement "div" {:className "test"}
                                                                             (js/React.createElement "span" nil "Test")
                                                                             (js/React.createElement "span" nil (str "Test" n))))))
                                        (::args state))
                                      state]) @state)]
             (vreset! state next-state)
             dom)))
```

So, I can't even imagine why this is happening.

**Comment by tonsky** [comment]
So key warning happens because sablono puts it into an array if component body does not start with vector. I’ll see if I can report it to them or work around it by myself

**Comment by sansolovyov** [comment]
It's reported at r0man/sablono#57

**Comment by quetz** [comment]
Is there any fix to this? I'm getting this with sablono 0.5.3 and rum 0.6.0
Everything is fine if I wrap contents of let-block with `(sablono.core/html ...)` but it looks like a workaround...

**Comment by r0man** [comment]
@tonsky This should be fixed in sablono `0.6.0`.

**Comment by tonsky** [comment]
Awesome! Will check tomorrow

**Comment by kennyjwilli** [comment]
Status update on this? Can we update sablono to 0.6.0?

**Comment by tonsky** [comment]
You can do it locally via :exclusions, it won’t break anything. Next
version of rum will also contain updated dependency

**Comment by tonsky** [comment]
`[sablono 0.6.3]` was included in `[rum 0.7.0]`, which should have this issue fixed
